### ISSUE TITLE
Linking Refprop runtime with CoolProp 

### ISSUE THREAD

**Issue by prat777** [CoolProp/REFPROP-headers#19]
Hello CoolProp Team,

I'm having trouble linking REFPROP-cmake with CoolProp. 

I followed all the instructions to build both CoolProp and REFPROP-cmake, from the following readme's without any errors. 

Using the 64-bit instructions
http://www.coolprop.org/coolprop/wrappers/SharedLibrary/index.html#shared-library
https://github.com/usnistgov/REFPROP-cmake

I added both CoolProp and REFPROP-cmake directories in a config file under the ld.so.conf.d directory, which are recognized by my system when using the ldconfig -p command. Where the following snippet are the relevant libraries in question. 
 
![image](https://user-images.githubusercontent.com/4643760/72366183-c3d30600-36c7-11ea-99b0-4364c280b03d.png)
![image](https://user-images.githubusercontent.com/4643760/72366248-e5cc8880-36c7-11ea-9ec5-5c327a35603d.png)

I found a sample c++ script from the CoolProp documentation (first link) and was able to compile and execute the original successfully, so I modified it slightly in hopes it would just link with REFPROP but I was wrong. 

```
#include <iostream>
#include "CoolPropLib.h"
#include "CoolProp.h"
#include "REFPROP_lib.h"

//Simple example
int main(){
    double T{293.15};
    double P{1e5};
    double res{CoolProp::PropsSI("H", "T", T, "P", P, "REFPROP::Water")};
    std::cout << "Enthalpy: " << res << std::endl;
}

```
I compiled the sample file using the following linkages which resulted in no runtime errors. The additional linkages are locations in CoolProp include and externals directories, including REFPROP-cmake's externals and build directories.  

```
g++ -std=c++11 -Wall -O2 -o snaple -DCOOLPROP_LIB -ICoolProp/include -ICoolProp/externals/fmtlib snaple.c -IREFPROP-cmake/externals/REFPROP-headers -IREFPROP-cmake/build CoolProp/build/libCoolProp.so.6 REFPROP-cmake/build/librefprop.so -ldl

```

When I run the compiled code, all I get is this. I know that the REFPROP 10.0.0.2 is 64 bit and I made sure everything is 64 bit. The paths are defined in the search but I am unsure whether I need to implement other parameters in this simple example code. 
 
```
Good news: It is possible to use REFPROP on your system! However, the library
could not be loaded. Please make sure that REFPROP is available on your system.

Neither found in current location nor found in system PATH.
If you already obtained a copy of REFPROP from http://www.nist.gov/srd/,
add location of REFPROP to the PATH environment variable or your library path.

In case you do not use Windows, have a look at https://github.com/jowr/librefprop.so
to find instructions on how to compile your own version of the REFPROP library.

ALTERNATIVE_REFPROP_PATH:
ERROR: Could not load REFPROP (librefprop.so) due to: /opt/refprop/librefprop.so: cannot open shared object file: No such file or directory. Make sure the library is in your system search path. In case you run 64bit Windows and you have a REFPROP license, try installing the 64bit DLL from NIST.
Enthalpy: inf

```

**Comment by ibell** [comment]
Try setting the path:
http://www.coolprop.org/coolprop/REFPROP.html#path-issues .  You'll note in
the error at the end that CoolProp tried to load from
/opt/refprop/librefprop.so.  CoolProp uses runtime linking, not
compile-time linking to REFPROP.

**Comment by prat777** [comment]
Compiler shoots out warning stating that set_config_string is not part of CoolProp? Suggests using set_warning_string()



Console message

![image](https://user-images.githubusercontent.com/4643760/72397982-6bbff200-370f-11ea-8930-08121790a125.png)


Attachment of simple code
```

#include <iostream>
#include <string>
#include "CoolPropLib.h"
#include "CoolProp.h"

//#define REFPROP_PROTOTYPES
//#define REFPROP_IMPLEMENTATION
#include "REFPROP_lib.h"
//#undef REFPROP_IMPLEMENTATION
//#undef REFPROP_PROTOTYPES

//Simple example


void setup() {
std::string str1 = "ALTERNATIVE_REFPROP_LIBRARY_PATH";
std::string str2 = "ALTERNATIVE_REFPROP_PATH";
std::string str3 = "ALTERNATIVE_REFPROP_HMX_BNC_PATH";


CoolProp::set_config_string(str1, "/home/bob/REFPROP-cmake/build/librefprop.so");
CoolProp::set_config_string(str2, "/home/bob/REFPROP-cmake");
CoolProp::set_config_string(str3, "/home/bob/REFPROP-cmake/FLUIDS/HMX.BNC");

}



int main(){

    //

    double T{293.15};
    double P{1e5};
    double res{CoolProp::PropsSI("H", "T", T, "P", P, "REFPROP::Water")};
    std::cout << "Enthalpy: " << res << std::endl;
}

```

**Comment by ibell** [comment]
You need to include "Configuration.h" header.  Also, please use the C++ tag in your code block (I fixed it, please see my edits).

**Comment by prat777** [comment]
Alright I added and linked Configuration.h

Now i'm looking at set_config_string() listed below, and I do not get why the compiler is displaying this message. I defined set_config_string() to take in two strings as specified in the document, but I don't get whey the function is trying to convert the string into a char in which it immediately fails due to the string being too long. 

http://www.coolprop.org/_static/doxygen/html/_configuration_8cpp_source.html

```C++
void setup() {
const std::string str1 = "ALTERNATIVE_REFPROP_LIBRARY_PATH";
const std::string str2 = "ALTERNATIVE_REFPROP_PATH";
const std::string str3 = "ALTERNATIVE_REFPROP_HMX_BNC_PATH";

const std::string str4 = {"/home/bob/REFPROP-cmake/build/librefprop.so"};
const std::string str5 = {"/home/bob/REFPROP-cmake"};
const std::string str6 = {"/home/bob/REFPROP-cmake/FLUIDS/HMX.BNC"};


set_config_string(str1, str4);
set_config_string(str2, str5);
set_config_string(str3, str6);

}


````
![image](https://user-images.githubusercontent.com/4643760/72399988-c8bea680-3715-11ea-8b9d-f10d254589b2.png)

**Comment by ibell** [comment]
The argument is not a string -- it is an enumerated value.  So
CoolProp::ALTERNATIVE_REFPROP_PATH

**Comment by prat777** [comment]
Thank you for the help,

With the REFPROP backend enabled PropSI returns a value of 'inf', and oddly enough when I check the Refprop version it returns a value of 10.0 instead of 10.0.0.2. I have all the paths defined to a remote location in my directory so I was wondering if i needed to initialize another function or set another path variable. I should have noted that I have different versions of REFPROP built onto the machine initially.

**Comment by ibell** [comment]
Did you end up setting all the path-related config variables on linux?  For me it is usually sufficient and most reliable to have the .so, the FLUIDS directory and the MIXTURES directory in the same location (this is how REFPROP is usually set up on windows), and then set the ALTERNATIVE_REFPROP_PATH to that folder.  If that doesn't work, then I suspect your REFPROP might be getting confused, and I would recommend to delete all traces of REFPROP from your machine and reinstall to make sure nothing funny is going on.  

Do you get any useful information if you enable debugging with set_debug_level(100) before you try your PropsSI call?  It will spit out rather a lot of info, but could help to get to the bottom of this.

**Comment by prat777** [comment]
Thank you so much ibell,
 using the debug level I learned that the backend was looking for lowercase directory names for fluids and mixtures, so I made the changes accordingly and the correct result were displayed. In addition I moved the fluids and mixtures directories to a common path that shares the librefprop.so. However I still find it odd that the REFPROP version identified is not 10.0.0.2, but that does not seem too relevant now.
